cibuildwheel/linux.py

```diff
@@ -158,11 +62,132 @@ def build(project_dir, output_dir, test_command, test_requires, test_extras, bef
                             '-i',
                             '-v', '/:/host',  # ignored on CircleCI
                             docker_image, '/bin/bash'], check=True)
-            subprocess.run(['docker', 'cp', os.path.abspath(project_dir) + '/.', container_name + ':/project'], check=True)
-            subprocess.run(['docker', 'start', '-i', '-a', container_name], input=bash_script, universal_newlines=True, check=True)
-            subprocess.run(['docker', 'cp', container_name + ':/output/.', os.path.abspath(output_dir)], check=True)
-        except subprocess.CalledProcessError as error:
-            troubleshoot(project_dir, error)
+            subprocess.run(['docker', 'cp',
+                            os.path.abspath(project_dir) + '/.',
+                            container_name + ':/project'], check=True)
+
+            for config in platform_configs:
+                if dependency_constraints:
+                    constraints_file = dependency_constraints.get_for_python_version(config.version)
+                    subprocess.run(['docker', 'cp',
+                                    os.path.abspath(constraints_file),
+                                    container_name + ':/constraints.txt'], check=True)
+
+                subprocess.run(
```

Comment:
Yes, just that and `pip install {dependency_install_flags} virtualenv` a bit further down

Reply:
> another PR, obviously, but did you still want to kill the bash script and replace it with Python, at some point

Yes, if it can be done nicely, I think it'd be an improvement. I still think the ideas in https://github.com/joerick/cibuildwheel/issues/99#issuecomment-420557347 have merit.

Reply:
Hmmmm, yes, true, I noticed that after writing the comment. But isn't it funny that there the constraints file is honoured, while for `pip` etc that're already installed in the manylinux image, it isn't?

Reply:
Yeah, it is a little. But really, the primary goal of this PR is determinism, and the virtualenv install is the only non-deterministic bit of the manylinux script.